docs/infrastructure/EKS/main/cluster_masters.tf

```diff
@@ -30,11 +30,46 @@ resource "aws_eks_cluster" "main" {
 
 }
 
+# Get an authentication token to communicate with an EKS cluster, using the "kubernetes" provider.
+data "aws_eks_cluster_auth" "eks_cluster_auth" {
```

Comment:
Please move data resources to the `constants.tf` file. This might lead to a cyclic dependency. This might not be needed. See below.